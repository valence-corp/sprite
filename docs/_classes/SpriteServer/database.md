---

layout: default

title: database

permalink: /classes/SpriteServer/database.html

---

### _SpriteServer_.database

#### Interface

(**databaseName: *string***)

Returns an SpriteDatabase client for the supplied `databaseName`,
using the authorization details of the `SpriteServer` client.

#### Example

```ts
const server = new SpriteServer({
  username: 'aUser',
  password: 'aPassword',
  address: 'http://localhost:2480',
});

const database = await server.database('aDatabase');
console.log(database.name);
// 'aDatabase';
```

