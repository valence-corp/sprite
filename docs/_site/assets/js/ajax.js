async function getDocument(url) {
  try {
    const response = await fetch(url);
    const html = await response.text();
    const parser = new DOMParser();
    const htmlDoc = parser.parseFromString(html, "text/html");
    const contentDiv = htmlDoc.getElementById("content");
    const contentHtml = contentDiv.outerHTML;
    document.getElementById("content").innerHTML = contentDiv.innerHTML;
  } catch (error) {
    throw new Error("Could not fetch page.", { cause: error });
  }
}

function createLinkClickEvents(root) {
  const elements = root.querySelectorAll("a");
  console.log(elements);
  if (elements.length) {
    elements.forEach((link) => {
      link.addEventListener("click", (e) => {
        e.preventDefault();
        getDocument(e.currentTarget.href);
      });
    });
  }
}

function ajax() {
  createLinkClickEvents(document);
}

document.addEventListener("DOMContentLoaded", ajax);
