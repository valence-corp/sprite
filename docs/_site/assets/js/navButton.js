document.addEventListener("DOMContentLoaded", function () {
  const navButton = document.getElementById("nav_button");
  const navIndicator = navButton.querySelector("svg");
  const mainNav = document.getElementById("mainNavigation");

  const closeIconMarkup = `
    <line x1="3" y1="4" x2="21" y2="21"></line>
    <line x1="21" y1="4" x2="3" y2="21"></line>
  `;

  const openIconMarkup = `
    <line x1="3" y1="4" x2="21" y2="4"></line>
    <line x1="3" y1="12" x2="21" y2="12"></line>
    <line x1="3" y1="20" x2="21" y2="20"></line>
  `;

  let isOpen = false;
  let previousTouchPosition = null;
  let currentTranslation = 0;

  function handleTouchEnd() {
    previousTouchPosition = null;

    if (currentTranslation > -80) {
      currentTranslation = 0;
      mainNav.style.transform = `translate3d(${currentTranslation}px, 0, 0)`;
    } else {
      mainNav.style.transform = "";
      currentTranslation = 0;
      closeMenu();
    }
  }

  function handleTouchStart(event) {
    previousTouchPosition = event.touches[0].clientX;
  }

  function handleTouchMove(event) {
    const currentTouchPosition = event.touches[0].clientX;
    const delta = currentTouchPosition - previousTouchPosition;
    previousTouchPosition = currentTouchPosition;

    currentTranslation += delta;

    if (currentTranslation > 0) {
      currentTranslation = 0;
    }

    mainNav.style.transform = `translate3d(${currentTranslation}px, 0, 0)`;
  }

  function enableTouchEvents() {
    mainNav.addEventListener("touchstart", handleTouchStart);
    mainNav.addEventListener("touchmove", handleTouchMove);
    mainNav.addEventListener("touchend", handleTouchEnd);
  }

  function disableTouchEvents() {
    mainNav.removeEventListener("touchstart", handleTouchStart);
    mainNav.removeEventListener("touchmove", handleTouchMove);
    mainNav.removeEventListener("touchend", handleTouchEnd);
  }

  function closeMenu() {
    toggleNavState();
    disableTouchEvents();
    navIndicator.innerHTML = openIconMarkup;
  }

  function openMenu() {
    toggleNavState();
    enableTouchEvents();
    navIndicator.innerHTML = closeIconMarkup;
  }

  function toggleNavState() {
    mainNav.classList.toggle("open");
    isOpen = !isOpen;
  }

  function handleNavButtonClick() {
    if (isOpen) {
      closeMenu();
    } else {
      openMenu();
    }
  }

  navButton.addEventListener("click", handleNavButtonClick);
});
