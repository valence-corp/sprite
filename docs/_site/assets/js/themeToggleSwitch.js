// function setTheme(theme) {
//   localStorage.setItem("theme", theme);
//   document.body.className = theme;
// }

// function setSwitchPosition(theme) {
//   switch (theme) {
//     case "dark":
//       themeToggleSwitch.checked = true;
//       break;
//     case "light":
//       themeToggleSwitch.checked = false;
//       break;
//     default:
//       throw new Error("unknow theme");
//   }
// }

// function setThemeAndSwitch(theme) {
//   setTheme(theme);
//   setSwitchPosition(theme);
// }

// function prefersTheme(theme) {
//   return window.matchMedia(`(prefers-color-scheme: ${theme})`).matches();
// }

// function initializeThemeToggle() {
//   const theme = localStorage.getItem("theme");
//   const toggleSwitch = document.getElementById("themeToggle");

//   toggleSwitch.addEventListener("click", () => {
//     if (themeToggleSwitch.checked) {
//       setTheme("dark");
//     } else {
//       setTheme("light");
//     }
//   });

//   if (theme) {
//     setThemeAndSwitch(theme);
//   } else {
//     if (prefersTheme("dark")) {
//       setThemeAndSwitch("dark");
//       return;
//     } else {
//       setThemeAndSwitch("light");
//     }
//   }
// }

// document.addEventListener("DOMContentLoaded", initializeThemeToggle);

class ThemeController {
  constructor(target) {
    this.element = target;
    this.theme = localStorage.getItem("theme");
    this.initialize();
  }
  initialize = () => {
    this.element.addEventListener("click", () => {
      if (this.element.checked) {
        this.changeTheme("dark");
      } else {
        this.changeTheme("light");
      }
    });
    this.determineInitialPosition();
  };
  prefersTheme = (theme) => {
    return window.matchMedia(`(prefers-color-scheme: ${theme})`).matches();
  };
  determineInitialPosition = () => {
    if (this.theme) {
      this.setTheme(this.theme);
    } else {
      if (prefersTheme("dark")) {
        this.setTheme("dark");
        return;
      } else {
        this.setTheme("light");
      }
    }
  };
  setPosition = (theme) => {
    switch (theme) {
      case "dark":
        this.element.checked = true;
        break;
      case "light":
        this.element.checked = false;
        break;
      default:
        throw new Error("unknow theme");
    }
  };
  changeTheme = (theme) => {
    localStorage.setItem("theme", theme);
    document.body.className = theme;
  };
  setTheme = (theme) => {
    this.changeTheme(theme);
    this.setPosition(theme);
  };
}

function initializeThemeToggle() {
  const element = document.getElementById("themeToggle");
  return new ThemeController(element);
}

document.addEventListener("DOMContentLoaded", initializeThemeToggle);
