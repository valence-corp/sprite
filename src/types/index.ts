export * from './client.js';
export * from './database.js';
export * from './edge.js';
export * from './server.js';
export * from './operators.js';
