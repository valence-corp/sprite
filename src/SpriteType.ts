import { SpriteCommand } from "./SpriteCommand.js";
import { SpriteDatabase } from "./SpriteDatabase.js";
import { SpriteTransaction } from "./SpriteTransaction.js";
import { createPropertyNodes } from "./nodes/create/property/index.js";
import { TypeNames } from "./types/database.js";
import {
  ArcadeSchemaConstraints,
  ArcadeSchemaDataType,
  SpriteSchemaDefinition,
} from "./types/type.js";

/**
 * @description Used to define properties of a type in the schema.
 * @experimental
 */
export class SpriteType<S, N extends TypeNames<S>> {
  private _database: SpriteDatabase;
  private _name: N;
  private _nodes = createPropertyNodes;
  /** The name of the type */
  get name() {
    return this._name;
  }
  constructor(database: SpriteDatabase, typeName: N) {
    this._name = typeName;
    this._database = database;
  }
  model = async (
    definitions: SpriteSchemaDefinition<S, N>,
    transaction: SpriteTransaction
  ) => {
    const properties = Object.keys(definitions) as [keyof typeof definitions];
    const reciept = [];
    for (let i = 0; i < properties.length; ++i) {
      const { type, ...constraints } = definitions[properties[i]];
      reciept.push(
        await this.createProperty(properties[i], type as any, transaction, {
          constraints: constraints as any,
        })
      );
    }
    return Promise.all(reciept);
  };
  createProperty = async <P extends keyof S[N]>(
    property: P,
    dataType: ArcadeSchemaDataType<S[N][P]>,
    transaction: SpriteTransaction,
    options?: {
      embeddedType?: S[N][P];
      constraints?: ArcadeSchemaConstraints<S[N][P]>;
      ifNotExists?: boolean;
    }
  ) => {
    const command = new SpriteCommand({
      initial: this._nodes.createProperty(
        this._name as string,
        property as string
      ),
    });

    if (options?.ifNotExists) {
      command.append<boolean>(this._nodes.ifNotExists, options.ifNotExists);
    }

    // TODO: command.append, perhaps it just appends the string
    // if that's all that's passed to it
    command.append((dataType) => dataType as string, dataType);

    if (options?.embeddedType) {
      command.append<S[N][P]>(
        (type: S[N][P]) => `OF ${type}`,
        options.embeddedType
      );
    }
    if (options?.constraints) {
      command.append<ArcadeSchemaConstraints<S[N][P]>>(
        this._nodes.constraints,
        options.constraints
      );
    }

    const response = await this._database.command<[{}]>(
      "sql",
      command.toString(),
      transaction
    );
    return response.result[0];
  };
}
