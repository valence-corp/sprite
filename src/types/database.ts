import { ArcadeSelectTimeoutStrategy } from '../nodes/types.js';
import { ISpriteRestClientConnectionParameters } from './client.js';
import { SpriteRestClient } from '../SpriteRestClient.js';
import { SpriteTransaction } from '../SpriteTransaction.js';
import { ValidSuperTypeKey } from '../SpriteType.js';

export type ArcadeRecordCategory = 'd' | 'e' | 'v';
export type ArcadeRecordType = 'document' | 'edge' | 'vertex';

export type RecordBase = {
  '@rid': string;
  '@cat': ArcadeRecordCategory;
  '@type': string;
};

export type EdgeBase = RecordBase & {
  '@in': string;
  '@out': string;
};

export type WithArcadeEdgeRecordMeta<S> = {
  [K in keyof S]: S[K] & EdgeBase;
};

export type WithArcadeRecordMeta<S> = {
  [K in keyof S]: S[K] & RecordBase;
};

export type TypeNames<Schema> = keyof Schema;

/**
 * The parameters necessary to perform opertations
 * on a target ArcadeDB database.
 */
export interface ISpriteDatabaseConnectionParameters
  extends ISpriteRestClientConnectionParameters {
  /**
   * The name of the database to connect to.
   */
  databaseName: string;
}

export interface ISpriteDatabaseClientParameters {
  client: SpriteRestClient;
  databaseName: string;
}

/**
 * The Query languages supported by ArcadeDB, to be declared when
 * using SpriteDatabase.query() and SpriteDatabase.command() methods
 */
export type ArcadeSupportedQueryLanguages =
  | 'sql'
  | 'sqlscript'
  | 'graphql'
  | 'cypher'
  | 'gremlin'
  | 'mongo';

/**
 * Options to create a new type with.
 */
export interface ISpriteCreateTypeOptions<S, N extends TypeNames<S>> {
  /**
   * When set to true, the type creation will be ignored if the
   * type already exists (instead of failing with an error).
   * @default false
   */
  ifNotExists?: boolean;
  /**
   * Defines a super-type you want to extend with this type.
   */
  extends?: ValidSuperTypeKey<S, N>;
  /**
   * A bucket-name, or an array of bucket-names you want this type to use.
   */
  buckets?: string | Array<string>;
  /**
   * Defines the total number of buckets you want to create for this type. The
   * @default 1
   */
  totalBuckets?: number;
}

export type OmitMeta<T> = Omit<T, keyof RecordBase | keyof EdgeBase>;

/** Options to insert a new record with */
export interface ISpriteInsertRecordOptions<T> {
  /** The data to populate the newly created reacord with */
  data?: OmitMeta<T>;
  /** The bucket to store the record in */
  bucket?: string;
}

/**
 * Options for a database.dropType() command
 */
export interface ISpriteDropTypeOptions {
  /**
   * Prevent errors if the type does not exits when attempting to drop it.
   * @default false;
   */
  ifExists?: boolean;
  /**
   * Defines whether the command drops non-empty edge and vertex types. Note, this can
   * disrupt data consistency. Be sure to create a backup before running it.
   * @default: false
   */
  unsafe?: boolean;
}

/** A function which contains the operations to be conducted within a transaction */
export type SpriteTransactionCallback =
  | ((trx: SpriteTransaction) => Promise<void>)
  | ((trx: SpriteTransaction) => void);

/**
 * How the bucket will be selected for a newly created record of this type
 * @default 'round-robin'
 */
export type ArcadeBucketSelectionStrategies =
  | 'round-robin'
  | 'thread'
  | 'partitioned<primary-key>';

/**
 * A type definition returned by ArcadeDB when a getSchema command
 * is compeleted.
 */
export type ArcadeTypeDefinition = {
  /** The name of the type */
  name: string;
  /** The category of the type (document, vertex, or edge) */
  type: ArcadeRecordType;
  /**
   * The number of records with this type name
   */
  records: number;
  /**
   * The buckets associated with this type.
   */
  buckets: Array<string>;
  /** How the bucket will be selected for a newly created record of this type */
  bucketSelectionStrategy: ArcadeBucketSelectionStrategies;
  /**
   * Super types associated with this type.
   */
  parentTypes: Array<string>;
  /**
   * Properties defined for this type.
   */
  properties: Array<string>;
  /** Indexes defined for records of this type */
  indexes: Array<string>;
  /** An object for custom user-defined properties for this type */
  custom: Record<string, unknown>;
};

export type ArcadeGetSchemaResponse = Array<ArcadeTypeDefinition>;

export type ArcadeSqlExplanationExecutionPlan = {
  type: 'QueryExecutionPlan' | 'DDLExecutionPlan';
  javaType: string;
  cost: number;
  prettyPrint: string;
  steps: object[];
};

/**
 * An object representing the explanation of an
 * SQL command.
 */
export type ArcadeSqlExplanation = {
  executionPlan: ArcadeSqlExplanationExecutionPlan;
  executionPlanAsString: string;
};

/**
 * An object returned from the server following a successfully command / query
 *
 */
export interface ArcadeCommandResponse<T = unknown> {
  /**
   * The user that issued the command / query.
   */
  user: string;
  /** The version of ArcadeDB returning the result. */
  version: string;
  /**
   * The name of the server returning the result.
   */
  serverName: string;
  /**
   * The result of the command / query.
   */
  result: T;
}

export type ArcadeQueryResponse<T = unknown> = ArcadeCommandResponse<T>;

// TODO: I have not checked these for compatibility with ArcadeDB
export declare const COMPARISON_OPERATORS: readonly [
  '=',
  '==',
  '!=',
  '<>',
  '>',
  '>=',
  '<',
  '<=',
  'in',
  'not in',
  'is',
  'is not',
  'like',
  'not like',
  'match',
  'ilike',
  'not ilike',
  '@>',
  '<@',
  '&&',
  '?',
  '?&',
  '!<',
  '!>',
  '<=>',
  '!~',
  '~',
  '~*',
  '!~*',
  '@@',
  '@@@',
  '!!',
  '<->',
  'regexp',
  'is distinct from',
  'is not distinct from',
];

export type SpriteOperators = (typeof COMPARISON_OPERATORS)[number];

export interface ISpriteSelectFromOptions<
  S,
  N extends keyof S,
  P extends keyof WithRid<S, N>,
> {
  /**
   * Designates conditions to filter the result-set.
   */
  where?: SpriteWhereClause<S, N, P>;
  /**
   * Designates the field with which to order the result-set.
   * Use the optional 'ASC' and 'DESC' operators to define the direction of the order.
   */
  orderBy?: {
    field: keyof S[N];
    /**
     * Defines the direction to sort the result (ASCending or DESCending).
     * @default 'ASC'.
     */
    direction?: 'ASC' | 'DESC';
  };
  /**
   * Defines the number of records you want to skip from the start of the result-set.
   * You mayfind this useful in Pagination, when using it in conjunction with the
   * limit `option`.
   */
  skip?: number;
  /**
   * Defines the maximum number of records in the result-set. You may find this useful in
   * Pagination, when using it in conjunction with the `skip` option.
   */
  limit?: number;
  /**
   * Defines the maximum time in milliseconds for the query, and optionally the
   * exception strategy to use.
   */
  timeout?: {
    /**
     *  The duration of the timeout in milliseconds.
     */
    duration: number;
    /**
     * The timeout strategy to use.\
     * `RETURN` Truncates the result-set, returning the data collected up to the timeout.\
     * `EXCEPTION` Raises an exception.
     */
    strategy?: ArcadeSelectTimeoutStrategy;
  };
}

export type WithRid<S, N extends TypeNames<S>> = OmitMeta<S[N]> & {
  '@rid': string;
};

export type SpriteWhereClause<
  S,
  N extends TypeNames<S>,
  P extends keyof WithRid<S, N>,
> = [P, SpriteOperators, WithRid<S, N>[P]];

export interface ISpriteDeleteFromOptions<
  S,
  N extends TypeNames<S>,
  P extends keyof WithRid<S, N>,
> {
  /**
   * Designates conditions to filter the result-set.
   */
  where: SpriteWhereClause<S, N, P>;
  /**
   * The duration of the timeout in milliseconds.
   */
  timeout?: number;
  /**
   * Defines the maximum number of records in the result-set.
   * @default undefined
   */
  limit?: number;
  /**
   * Defines what is returned following the command: the count of the records before (`BEFORE`) or following deletion (`COUNT`).
   * @default 'COUNT'
   */
  return?: 'COUNT' | 'BEFORE';
}
