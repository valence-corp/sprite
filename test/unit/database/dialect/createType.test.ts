import { client } from './testClient.js';
import { endpoints } from '../../../../src/endpoints/database.js';
import { variables, headers } from '../../../variables.js';
import { SpriteType } from '../../../../src/SpriteType.js';

import { DocumentTypes } from '../../types.js';

const recordType = 'document';
const typeName = 'aDocument';
const SpriteDatabase = client.database;
type TypeName = typeof typeName;

const createTypeResult = {
  user: variables.username,
  serverName: '',
  version: '',
  result: [{ typeName }]
};

describe('TypedOperations.createType()', () => {
  it(`makes a properly formatted POST request to ${endpoints.command}/${variables.databaseName}`, async () => {
    // Arrange
    jest.spyOn(global, 'fetch').mockResolvedValueOnce({
      status: 200,
      json: async () => createTypeResult
    } as Response);

    // Act
    await client.createType<DocumentTypes, TypeName>(typeName, recordType);
    // Assert
    expect(global.fetch).toHaveBeenCalledWith(
      `${variables.address}${endpoints.command}/${variables.databaseName}`,
      {
        method: 'POST',
        headers,
        body: JSON.stringify({
          language: 'sql',
          command: `CREATE document TYPE ${typeName}`
        })
      }
    );
  });

  it('it returns an instance of SpriteType when it receives a a JSON object with a result property from the server.', async () => {
    // TODO: This seems confusing.
    // reasoning for this is because the if it's passed a
    // 'ifNotExists' option, it won't throw an error, and
    // the user might be expecting a 'type' to perform
    // operations on, so we return the type even if
    // it was previously created.

    // Arrange
    jest
      .spyOn(SpriteDatabase, 'command')
      .mockResolvedValueOnce(createTypeResult);

    // Act
    const result = await client.createType<DocumentTypes, TypeName>(
      typeName,
      recordType,
      {
        buckets: variables.bucketName
      }
    );
    // Assert
    expect(result).toBeInstanceOf(SpriteType);
  });

  it('handles "buckets" option by appending "BUCKET" + [bucketName] to the command when passed a string bucketName property.', async () => {
    // Arrange
    jest
      .spyOn(SpriteDatabase, 'command')
      .mockResolvedValueOnce(createTypeResult);

    // Act
    await client.createType<DocumentTypes, TypeName>(typeName, recordType, {
      buckets: variables.bucketName
    });

    // Assert
    expect(SpriteDatabase.command).toHaveBeenCalledWith(
      'sql',
      `CREATE document TYPE ${typeName} BUCKET ${variables.bucketName}`
    );
  });

  it('handles "buckets" option by appending "BUCKET" + [bucketNames] to the command when passed an array of strings as the bucketName property.', async () => {
    // Arrange
    jest
      .spyOn(SpriteDatabase, 'command')
      .mockResolvedValueOnce(createTypeResult);

    // Act
    await client.createType<DocumentTypes, TypeName>(typeName, recordType, {
      buckets: [variables.bucketName, variables.bucketName]
    });

    // Assert
    expect(SpriteDatabase.command).toHaveBeenCalledWith(
      'sql',
      `CREATE document TYPE ${typeName} BUCKET ${[
        variables.bucketName,
        variables.bucketName
      ].join(',')}`
    );
  });

  it('handles "ifNotExists" option by appending "IF NOT EXISTS" to the command when passed "false"', async () => {
    // Arrange
    jest
      .spyOn(SpriteDatabase, 'command')
      .mockResolvedValueOnce(createTypeResult);

    // Act
    await client.createType<DocumentTypes, TypeName>(typeName, recordType, {
      ifNotExists: true
    });

    // Assert
    expect(SpriteDatabase.command).toHaveBeenCalledWith(
      'sql',
      `CREATE document TYPE ${typeName} IF NOT EXISTS`
    );
  });

  it('handles "extends" option by appending "EXTENDS" + [typeName] to the command when passed an extends property', async () => {
    // Arrange
    jest
      .spyOn(SpriteDatabase, 'command')
      .mockResolvedValueOnce(createTypeResult);

    // Act
    await client.createType<DocumentTypes, TypeName>(typeName, recordType, {
      extends: 'anotherDocument'
    });

    // Assert
    expect(SpriteDatabase.command).toHaveBeenCalledWith(
      'sql',
      `CREATE document TYPE ${typeName} EXTENDS anotherDocument`
    );
  });

  it('handles "totalBuckets" option by appending "BUCKETS" + [totalBuckets] to the command when passed a totalBuckets property', async () => {
    // Arrange
    jest
      .spyOn(SpriteDatabase, 'command')
      .mockResolvedValueOnce(createTypeResult);

    // Act
    await client.createType<DocumentTypes, TypeName>(typeName, recordType, {
      totalBuckets: 4
    });
    // Assert
    expect(SpriteDatabase.command).toHaveBeenCalledWith(
      'sql',
      `CREATE document TYPE ${typeName} BUCKETS 4`
    );
  });

  it('should throw an error if it receives an empty string for parameters', async () => {
    //@ts-expect-error - Testing for empty string
    await expect(client.createType('')).rejects.toMatchSnapshot();
  });

  it('should throw an error if it receives no arguments', async () => {
    //@ts-expect-error - Testing for no arguments
    await expect(client.createType()).rejects.toMatchSnapshot();
  });
});
