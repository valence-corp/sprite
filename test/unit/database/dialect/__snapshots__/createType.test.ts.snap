// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypedOperations.createType() should throw an error if it receives an empty string for parameters 1`] = `[Error: Unable to create type: [''], in database: ['Test']]`;

exports[`TypedOperations.createType() should throw an error if it receives no arguments 1`] = `[Error: Unable to create type: ['undefined'], in database: ['Test']]`;
